Sources/TuistSupport/Extensions/AbsolutePath+Extras.swift

```diff
@@ -82,16 +82,19 @@ extension AbsolutePath {
     public var isInOpaqueDirectory: Bool {
         var currentDirectory = parentDirectory
         while currentDirectory != .root {
-            if let `extension` = currentDirectory.extension,
-               Self.opaqueDirectoriesExtensions.contains(`extension`)
-            {
-                return true
-            }
+            if currentDirectory.isOpaqueDirectory { return true }
             currentDirectory = currentDirectory.parentDirectory
         }
         return false
     }
 
+    /// An opaque directory is a directory that should be treated like a file, therefor ignoring its content.
+    /// I.e.: .xcassets, .xcdatamodeld, etc...
+    /// This property returns true when a file is such a directory.
+    public var isOpaqueDirectory: Bool {
+        Self.opaqueDirectoriesExtensions.contains(self.extension ?? "")
```

Comment:
```suggestion
        [
          "xcassets",
          "scnassets",
          "xcdatamodeld",
          "docc",
          "playground",
          "bundle",
          "mlmodelc",
      ]
      .contains(self.extension ?? "")
```

Reply:
Good point, removed `opaqueDirectoriesExtensions` and integrated it into `isOpaqueDirectory`